Library/Homebrew/utils/shell_env.sh

```diff
@@ -0,0 +1,71 @@
+keepkeys=(
+HOMEBREW_*
+GEM_*
+HOME
+PWD
+LANG
+PATH
+)
+
+prune-shell-env() {
+  # Associative arrays are only a Bash 4 feature
+  #declare -A aEnv
+
+  local verbose
+  
+  local envPATH=`which env`
+  local echoPATH=`which echo`
+  local sedPATH=`which sed`
+
+  file=$HOMEBREW_LIBRARY/Homebrew/utils/shell-env.txt
+  if [[ -n "$HOMEBREW_VERBOSE" ]]
+  then
+    $echoPATH 
+    $echoPATH "---- ENV pruning ----"
+    $echoPATH "Writing current env to '"$file"'"
+    # Need to keep a local record of 'HOMEBREW_VERBOSE=1' since will be temp unset
+    verbose=1
+  fi
+  
+  $envPATH > $file
+  
+  unset `$envPATH | $sedPATH -e 's/=.*//'`
+  
+  for i in ${keepkeys[*]}
+  do
+  
+    if [[ -n "$verbose" ]]
+    then
+      $echoPATH "Keeping: "$i
+    fi
+    
+    IFS="="
+    while read key value
+    do
+      if [[ $key == $i ]]
+      then
+      
+        if [[ -n "$verbose" ]]
+        then
+          $echoPATH $key"="$value
+        fi
+
+        export $key"="$value
+      fi
+    done < "$file"
+
+  done
+
+  pfile=$HOMEBREW_LIBRARY/Homebrew/utils/shell-env-pruned.txt
+  if [[ -n "$verbose" ]]
+  then
+    $echoPATH "Writing new env to '"$pfile"'"
+    $echoPATH "---------------------"
+    $echoPATH
+  fi
```

Comment:
Partly a dev feature left in and maybe the pruned path could be used in other homebrew commands like 'doctor'.  Happy to remove it if you want.